runtime/databricks/automl_runtime/forecast/prophet/model.py

```diff
@@ -117,35 +136,44 @@ def predict_timeseries(self, horizon: int = None) -> pd.DataFrame:
         """
         return self._predict_impl(horizon)
 
-    def predict(self, context: mlflow.pyfunc.model.PythonModelContext, input: pd.DataFrame) -> pd.DataFrame:
+    def predict(self, context: mlflow.pyfunc.model.PythonModelContext, model_input: pd.DataFrame) -> pd.DataFrame:
         """
         Predict API from mlflow.pyfunc.PythonModel
         :param context: A :class:`~PythonModelContext` instance containing artifacts that the model
                         can use to perform inference.
-        :param input: Input dataframe
+        :param model_input: Input dataframe
         :return: A pd.DataFrame with the forecast components.
         """
-        return self._predict_impl()
+        self._validate_cols(model_input, [self._time_col])
+        model_input.rename(columns={self._time_col: "ds"}, inplace=True)
```

Comment:
Instead of rename, let's do `X = pd.DataFrame{"ds": model_input[self._time_col]}`, so that the model input is not modified?

Reply:
Since we didn't return model_input, the user will not see the changes, right?

Reply:
No, python function doesn't do pass-parameter-by-value. Here's an example

```
def func(df):
  df.rename(columns:{"foo", "bar"}, inplace=True)

df = pd.DataFrame({"foo": [1,2, 3,4])
df

```